css/src/components/hero.scss

```diff
@@ -0,0 +1,153 @@
+$hero-padding-default: $layout-6 !default;
+$hero-padding-lg: $layout-7 !default;
+
+$hero-min-height-xs: 200px !default;
+$hero-min-height-sm: 300px !default;
+$hero-min-height-default: 400px !default;
+$hero-min-height-lg: 500px !default;
+
+$hero-content-width: $reading-max-width !default;
+$hero-content-fluid-width: 45% !default;
+$hero-image-width: calc((100% - 2 * $layout-gap) * 0.55) !default;
+$hero-image-width-widescreen: calc((100% - 2 * $layout-widescreen-gap) * 0.55) !default;
+
+.hero {
+	@extend %layout-gap;
+
+	display: flex;
+	position: relative;
+	flex-direction: column;
+	align-items: stretch;
+	justify-content: space-between;
+
+	.hero-content {
+		z-index: $zindex-active;
+		padding-block: $hero-padding-sm;
+
+		@include desktop {
+			width: $hero-content-fluid-width;
+			min-height: $hero-min-height-default;
+			padding-block: $hero-padding-default;
+			padding-inline-end: $hero-padding-sm;
+		}
+	}
+
+	&:not(.hero-image) {
+		.hero-content {
+			@include desktop {
+				width: 100%;
+				max-width: $hero-content-width;
+			}
+		}
+	}
+
+	@include desktop {
+		&.hero-xs {
+			.hero-content {
+				min-height: $hero-min-height-xs;
+				padding-block: $hero-padding-xs;
+			}
+		}
+
+		&.hero-sm {
+			.hero-content {
+				min-height: $hero-min-height-sm;
+				padding-block: $hero-padding-sm;
+			}
+		}
+
+		&.hero-lg {
+			.hero-content {
+				min-height: $hero-min-height-lg;
+				padding-block: $hero-padding-lg;
+			}
+		}
+	}
+
+	.hero-card {
+		position: relative;
+		padding: $spacer-5;
+		border: none;
+		background-color: transparent;
+		z-index: 1;
+
+		@include tablet {
+			padding: $spacer-10;
+		}
+
+		@include desktop {
+			border-right: 1px solid $border;
+			border-bottom: 1px solid $border;
+			border-left: 1px solid $border;
+			border-bottom-left-radius: $border-radius-lg;
+			border-bottom-right-radius: $border-radius-lg;
+		}
+	}
+
+	.hero-background {
+		position: absolute;
+		top: 0;
+		right: 0;
+		bottom: 0;
+		left: 0;
+		width: 100%;
+		height: 100%;
+		overflow: hidden;
+
+		video,
+		img {
+			width: 100%;
+			height: 100%;
+			object-fit: cover;
+		}
+	}
+}
+
+.hero-image {
+	&::before {
+		display: none;
+		position: absolute;
+		top: 0;
+		width: $hero-image-width;
+		height: 100%;
+		background-image: var(--hero-background-image);
+		background-repeat: no-repeat;
+		background-size: cover;
+		content: ' ';
+		inset-inline-end: $layout-gap;
+
+		@include desktop {
+			display: block;
+		}
+
+		@include widescreen {
+			width: $hero-image-width-widescreen;
+			inset-inline-end: $layout-widescreen-gap;
+		}
+
+		@media (forced-colors: active) {
+			background-image: none !important;
+		}
+
+		.theme-light & {
+			--hero-background-image: var(--hero-background-image-light);
+		}
+
+		.theme-dark & {
+			--hero-background-image: var(--hero-background-image-dark);
+		}
+
+		.theme-high-contrast & {
+			--hero-background-image: none;
+		}
+	}
+
+	&.hero-image-gradient-accent-right {
```

Comment:
Yikes on the name, can put this in a new file or something?

Reply:
Or maybe we need to do some more CSS variable trickery here? Would require some other extensions, but at least we shouldn't be tied necessarily to this one color.

```scss
background-image: background-image: linear-gradient(90deg, transparent 90%, var(--hero-gradient-color), 100%),
				var(--hero-background-image);
```

Then you'd have

```scss
.hero-gradient-color-accent {
   $body-background-accent
}
```

Reply:
Any suggestions on a better name? I also feel like it has to live in this file because it is tightened to the hero image

Reply:
Will have to spend some time thinking this one over, seems like we've got a single use class in here, which isn't good. It only works with one background color and doesn't seem extensible.